src/main/java/eu/europa/ted/eforms/sdk/repository/SdkCodelistRepository.java

```diff
@@ -63,125 +71,170 @@ public SdkCodelistRepository(String sdkVersion, Path codelistsDir) {
+    if (codelistId == null) {
+      return null;
     }
 
-    return computeIfAbsent((String) codelistId, key -> {
-      try {
-        return loadSdkCodelist(sdkVersion, key);
-      } catch (InstantiationException e) {
-        throw new RuntimeException(e);
-      }
-    });
+    return computeIfAbsent((String) codelistId,
+        Unchecked.function((String key) -> loadSdkCodelist(key).orElse(null)));
   }
 
+  @Nullable
   @Override
   public SdkCodelist getOrDefault(final Object codelistId, final SdkCodelist defaultValue) {
-    SdkCodelist result = get(codelistId);
-    return result != null ? result : defaultValue;
+    return Optional.ofNullable(get(codelistId)).orElse(defaultValue);
   }
 
-  private SdkCodelist loadSdkCodelist(final String sdkVersion, final String codeListId)
+  private Optional<SdkCodelist> loadSdkCodelist(@Nullable final String codeListId)
       throws InstantiationException {
     logger.debug("Loading SDK codelist with ID [{}] for SDK version [{}] from path [{}]",
         codeListId, sdkVersion, codelistsDir);
 
     // Find the SDK codelist .gc file that corresponds to the passed reference.
     // Stream the data from that file.
-    final Genericode10CodeListMarshaller marshaller = GenericodeTools.getMarshaller();
-
-    final Path filepath = codelistFilesByCodelistId.get(codeListId);
-    assert filepath != null : "filepath is null";
-
-    try (InputStream is = Files.newInputStream(codelistsDir.resolve(filepath))) {
-      final CodeListDocument codelist = marshaller.read(is);
-
-      // Get all the code values in a list.
-      // We assume there are no duplicate code values in the referenced
-      // codelists.
-      final List<String> codes = codelist
-          .getSimpleCodeList()
-          .getRow().stream()
-          .map(row -> row.getValue().stream()
-              .filter(v -> GenericodeTools.KEY_CODE.equals(GenericodeTools.extractColRefId(v)))
-              .findFirst()
-              .orElseThrow(RuntimeException::new)
-              .getSimpleValue()
-              .getValue()
-              .strip())
-          .collect(Collectors.toList());
-
-      // Version tag of the genericode (gc) file.
-      final String codelistVersion = codelist.getIdentification().getVersion();
-
-      final Optional<String> parentId = extractParentId(codelist.getIdentification());
-
-      SdkCodelist result =
-          SdkEntityFactory.getSdkCodelist(sdkVersion, codeListId, codelistVersion, codes,
-              parentId);
-
-      logger.debug("Finished loading SDK codelist with ID [{}] for SDK version [{}] from path [{}]",
-          codeListId, sdkVersion, codelistsDir);
-
-      return result;
-    } catch (final IOException e) {
-      throw new RuntimeException(e);
+    final Optional<CodeListDocument> codelist =
+        Optional.ofNullable(codelistContentsByCodelistId.get(codeListId));
+
+    if (codelist.isEmpty()) {
+      return Optional.empty();
```

Comment:
I don't know the business logic here.
Just kept the one that was already there.
If you tell me that a missing codelist is not a valid scenario, then sure, let's throw an exception.
But it would be something new for this code.